ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionLayerChannelImpl.java

```diff
@@ -213,39 +224,61 @@ public SafeFuture<TransitionConfiguration> engineExchangeTransitionConfiguration
   }
 
   @Override
-  public SafeFuture<ExecutionPayloadHeader> getPayloadHeader(
+  public SafeFuture<ExecutionPayloadHeader> builderGetHeader(
       final Bytes8 payloadId, final UInt64 slot) {
```

Comment:
My guess is that we need to start passing the data this method needs in rather than just payloadId. I think `ForkChoiceNotifier.getPayloadId` is already being called on the beacon node side before it calls through to this method and it knows the proposer ID (which can be mapped to validator ID easily) and the parent execution block hash. I think it's even verifying they're the right ones for the request so the payloadID matches up so it should fit pretty well to do the work there.  Might be some complications I've missed though.